iroha/src/torii.rs

```diff
@@ -182,41 +186,20 @@ async fn handle_queries(
     request: HttpRequest,
 ) -> Result<HttpResponse, String> {
     //TODO: Remove when `Result::flatten` https://github.com/rust-lang/rust/issues/70142 will be stabilized
-    let range = match Pagination::try_from(&query_params) {
-        Ok(range) => range,
-        Err(e) => {
-            log::error!("Failed to decode pagination: {}", e);
-            return Ok(HttpResponse::internal_server_error());
-        }
-    };
-
-    let request = match SignedQueryRequest::try_from(request.body) {
-        //TODO: check query permissions based on signature?
-        Ok(request) => request,
-        Err(e) => {
-            log::error!("Failed to decode transaction: {}", e);
-            return Ok(HttpResponse::internal_server_error());
-        }
-    };
-    let request = match VerifiedQueryRequest::try_from(request) {
-        Ok(request) => request,
-        Err(e) => {
-            log::error!("Failed to verify Query Request: {}", e);
-            return Ok(HttpResponse::internal_server_error());
-        }
-    };
-
-    let result = match request
+    let range = Pagination::try_from(&query_params)
+        .map_err(|e| format!("Failed to decode pagination: {}", e))?;
```

Comment:
Is it okay to return errors here instead of response?

Reply:
Yes, for http endpoints' handlers returning an error is the same as returning Internal Sever Error and logging the error message. For the internal networking protocol it is not the same.

In general we need to work on this and create an abstraction over errors, so that we will return HttpError as Result::Err and this will be easier to understand.